### ISSUE TITLE
Router plugin leads to erased query params on load

### ISSUE THREAD

**Issue by fabioscala** [ngxs/store#1153]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => https://github.com/ngxs/store/blob/master/CONTRIBUTING.md
[ ] Other... Please describe:
</code></pre>

## Current behavior
When the user enters an URL containing query parameters, the router plugin will redirect to the same path without query parameters. I was able to trace it down to this line:

https://github.com/ngxs/store/blob/7b0d9951db4f614b26b162633c968cbe62796441/packages/router-plugin/src/router.state.ts#L210

Without reading too much of the code, it seems that if statement compares the URL from the angular router (which contains query params) with the URL without it. The `path()` method used currently includes a boolean flag though:

```
    // TODO: vsavkin. Remove the boolean flag and always include hash once the deprecated router is
    // removed.
    path(includeHash = false)
```

## Expected behavior
No to be redirected and erase all the original query parameters

## Environment

<pre><code>
Libs:
- @angular/core version: 8.1.0
- "@ngxs/devtools-plugin": "3.4.3-dev.master-e7f64b9",
- "@ngxs/logger-plugin": "3.4.3-dev.master-7b0d995",
- "@ngxs/router-plugin": "3.4.3-dev.master-e7f64b9",
- "@ngxs/storage-plugin": "3.4.3-dev.master-e7f64b9",
- "@ngxs/store": "3.4.3-dev.master-7b0d995",

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by splincode** [comment]
Please check dev version:

```
$ npm I @ngxs/{store,router-plugin}@dev
```

is there still a problem?

**Comment by fabioscala** [comment]
I am already using the latest/nightly version of all, updated as of yesterday

**Comment by splincode** [comment]
add stackblitz example for reproduce

**Comment by fabioscala** [comment]
Here you go: https://stackblitz.com/edit/ngxs-simple-c6zex5

However after debugging the minimal sample, I was slightly wrong. That `if` check mentioned above is triggered because the lefthand `url` is escaped, while the the right side is not.

So the issue arises with initial URLs like this:
https://ngxs-simple-c6zex5.stackblitz.io/test?redirect=https://www.google.com/

The param `redirect` will be escaped on one side `/test?redirect=https:%2F%2Fwww.google.com%2F` while it's not on the other `/test?redirect=https://www.google.com/`.

**Comment by splincode** [comment]
@arturovt what do you think about this bug?

**Comment by arturovt** [comment]
I will look in the evening or later.

**Comment by arturovt** [comment]
@fabioscala Could you install the latest `@dev` version and try it out?

**Comment by markwhitfeld** [comment]
I just tried the stackblitz example (which didn't have much in it), upgraded to the latest, and it avoids the redirect with the provided url example. Win!

**Comment by arturovt** [comment]
> I just tried the stackblitz example (which didn't have much in it), upgraded to the latest, and it avoids the redirect with the provided url example. Win!

:tada: :tanabata_tree:

**Comment by fabioscala** [comment]
> @fabioscala Could you install the latest `@dev` version and try it out?

Works 👍 thanks

**Comment by splincode** [comment]
Released in 3.5.0
